OpenCL/m12150-pure.cl

```diff
@@ -0,0 +1,139 @@
+/**
+ * Author......: See docs/credits.txt
+ * License.....: MIT
+ */
+
+#define NEW_SIMD_CODE
+
+#ifdef KERNEL_STATIC
+#include M2S(INCLUDE_PATH/inc_vendor.h)
+#include M2S(INCLUDE_PATH/inc_types.h)
+#include M2S(INCLUDE_PATH/inc_platform.cl)
+#include M2S(INCLUDE_PATH/inc_common.cl)
+#include M2S(INCLUDE_PATH/inc_simd.cl)
+#include M2S(INCLUDE_PATH/inc_hash_sha512.cl)
+#endif
+
+#define COMPARE_S M2S(INCLUDE_PATH/inc_comp_single.cl)
+#define COMPARE_M M2S(INCLUDE_PATH/inc_comp_multi.cl)
+
+#define SHA512_DIGEST_LENGTH 64
+
+typedef struct shiro1_sha512
+{
+  u32 salt_buf[4];
+  u32 iterations;
+} shiro1_sha512_t;
+
+typedef struct shiro1_sha512_tmp
+{
+  u64 dgst[8];

```

Comment:
Why it is `u64`? Cannot it be `u32`? Conversions from `u64` to `u32` won't be necessary later.